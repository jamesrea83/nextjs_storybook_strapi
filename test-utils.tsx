import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';

import { Themes } from './styles/themes';

interface Props {
	children: ReactNode;
}

const Wrapper = ({ children }: Props) => (
	<ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
	render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react';

export { customRender as render };
