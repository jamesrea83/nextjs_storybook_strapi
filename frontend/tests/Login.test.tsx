import { render, screen, act } from '@/test-utils';
import userEvent from '@testing-library/user-event';

import Login from '@/pages/login';

describe('Login page', () => {
	it('should render', () => {
		const { container } = render(<Login />);
		expect(container).toMatchSnapshot();
	});

	it('should validate input', async () => {
		render(<Login />);

		const identifierInput = screen.getByRole('textbox', { name: 'Identifier' });
		const passwordInput = screen.getByRole('textbox', { name: 'Password' });
		const submitButton = screen.getByRole('button', { name: 'Sign in' });

		await act(() => userEvent.click(submitButton));

		expect(screen.getAllByText('Required field')).toHaveLength(2);

		await act(() => userEvent.type(identifierInput, 'test'));

		expect(screen.getByText('Min length 6!')).toBeInTheDocument();
		expect(screen.getByText('Required field')).toBeInTheDocument();

		await act(() => userEvent.type(passwordInput, 'test'));

		expect(screen.getAllByText('Min length 6!')).toHaveLength(2);
		expect(screen.queryByText('Required field')).not.toBeInTheDocument();

		await act(() => userEvent.type(identifierInput, 'test@test.com'));
		await act(() => userEvent.type(passwordInput, 'testtesttest'));

		await expect(screen.queryByText('Min length 6!')).not.toBeInTheDocument();
	});
});
