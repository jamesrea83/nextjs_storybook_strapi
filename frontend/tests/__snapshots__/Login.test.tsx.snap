// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page should render 1`] = `
<div>
  <form>
    <div
      class="css-42099h"
    >
      <section
        class="css-1beoq7g"
      >
        <h2>
          Login
        </h2>
        <label
          class="css-fp4izo"
          height="8"
          width="20"
        >
          <span
            class="css-porbx7"
          >
            Identifier
          </span>
          <div
            class="css-bnug8b"
          >
            <input
              class="css-58wzc"
              name="identifier"
              placeholder="username or email"
              role="textbox"
            />
          </div>
          <span
            class="css-tx5ech"
          >
             
          </span>
        </label>
        <label
          class="css-fp4izo"
          height="8"
          width="20"
        >
          <span
            class="css-porbx7"
          >
            Password
          </span>
          <div
            class="css-bnug8b"
          >
            <input
              class="css-58wzc"
              name="password"
              placeholder="password"
              role="textbox"
              type="password"
            />
          </div>
          <span
            class="css-tx5ech"
          >
             
          </span>
        </label>
        <button
          class="css-odwgcq"
          color="primary"
          type="submit"
        >
          Sign in
        </button>
        <h3>
          <a
            class="css-yf649i"
            href="/registration"
            underline="1"
          >
            Create account
          </a>
        </h3>
      </section>
    </div>
  </form>
</div>
`;
